### PR TITLE
feat: upload image

### PR DESCRIPTION
**Pull request by sshanzel**
We currently don't have any test for uploading to cloudinary, hence, no new test cases.

The concerns for optimizations on storage and for uploading GIFs, will be worked on separately. This is to unblock creating the freeform page and allow uploading for development.

### CODE REVIEW

File: src/schema/posts.ts [hunk 1]
```diff
@@ -874,6 +885,24 @@ export const resolvers: IResolvers<any, Context> = {
...
+
+      if (!process.env.CLOUDINARY_URL) {
+        throw new Error('Unable to upload asset to cloudinary!');
+      }
+
+      const upload = await image;
+      const id = generateShortId();
+      const filename = `post_content_${id}`;
+
+      return uploadPostImage(filename, upload.createReadStream());
```
[the remark this answers is not in the document]
Comment (sshanzel) [reply src/schema/posts.ts:904]: Awesome. I'll apply the rate-limiting now while we verify the correct preset.

**Comment by idoshamun** [comment]
@sshanzel Please use `freeform_image` upload preset. I'm also checking GIF optimizations

**Review by rebelchris — APPROVED** [review]
Looks neat, nice work on uniforming the types.
Non blocking from my side.

**Review by idoshamun — APPROVED** [review]
Awesome!

**Pull request merged**